include/util/mmap_file.hpp

```diff
@@ -33,6 +33,32 @@ util::vector_view<T> mmapFile(const boost::filesystem::path &file, RegionT &regi
             SOURCE_REF);
     }
 }
+
+template <typename T, typename RegionT>
+util::vector_view<T>
+mmapFile(const boost::filesystem::path &file, RegionT &region, const std::size_t size)
+{
+    try
+    {
+        // Create a new file with the given size in bytes
+        boost::iostreams::mapped_file_params params;
+        params.path = file.c_str();
+        params.mode = std::ios::in | std::ios::out;
```

Comment:
`mode` is deprecated, better to use `params.flags = boost::iostreams::mapped_file::readwrite;`